x-pack/plugins/security_solution/scripts/endpoint/resolver_generator_script.ts

```diff
@@ -37,6 +37,61 @@ async function deleteIndices(indices: string[], client: Client) {
   }
 }
 
+interface UserInfo {
+  username: string;
+  password: string;
+}
+
+async function addUser(
+  esClient: Client,
+  user?: { username: string; password: string }
+): Promise<UserInfo | undefined> {
+  if (!user) {
+    return;
+  }
+
+  const path = `_security/user/${user.username}`;
+  // add user if doesn't exist already
+  try {
+    console.log(`Adding ${user.username}...`);
+    const addedUser = await esClient.transport.request<Promise<{ created: boolean }>>({
+      method: 'POST',
+      path,
+      body: {
+        password: user.password,
+        roles: ['superuser', 'kibana_system'],
+        full_name: user.username,
+      },
+    });
+    if (addedUser.created) {
+      console.log(`User ${user.username} added successfully!`);
+    } else {
+      console.log(`User ${user.username} already exists!`);
+    }
+    return {
+      username: user.username,
+      password: user.password,
+    };
+  } catch (error) {
+    handleErr(error);
+  }
+}
+
+async function deleteUser(esClient: Client, username: string): Promise<{ found: boolean }> {
+  return esClient.transport.request({
+    method: 'DELETE',
+    path: `_security/user/${username}`,
+  });
+}
+
+const updateWithNewCredentials = (
+  url: string[],
+  user: { username: string; password: string }
+): string => {
+  url.splice(0, 1, `http://${user.username}:${user.password}`);
```

Comment:
Suggesiton: you probably should use `URL` class:

```typescript
const newUrl  = new URL(existingUrlString);

newUrl.username = user.username;
newUrl.password = use.password;

return newUrl.href

```

There might be other areas below where the use of `URL` might be best rather than trying to parse out the string.

Reply:
Nice! I'll use `URL` class instead! 🎉

Reply:
done 88ac44112c1500dfb558d5efd0d05f2a0e8983d2

Reply:
also 06eec6a5b009ad5e8f806ee6554ada1af0d1bb54